Raven.Database/Bundles/Replication/Impl/ReplicationHiLo.cs

```diff
@@ -16,52 +16,59 @@ namespace Raven.Bundles.Replication.Impl
 	internal class ReplicationHiLo
 	{
 		private readonly object generatorLock = new object();
-		private volatile Hodler currentMax = new Hodler(0);
+		private volatile ValueHolder valueHolder = new ValueHolder(0);
 		private long capacity = 256;
-		private long current;
 		private DateTime lastRequestedUtc;
 		public DocumentDatabase Database { get; set; }
 
-		private class Hodler
-		{
-			public readonly long Value;
+        public class ValueHolder
+        {
+            public readonly long MaxValue;
 
-			public Hodler(long value)
-			{
-				Value = value;
-			}
-		}
+            public long Current;
 
-		public long NextId()
-		{
-			long incrementedCurrent = Interlocked.Increment(ref current);
-			while (incrementedCurrent > currentMax.Value)
-			{
-				lock (generatorLock)
-				{
-					if (current > currentMax.Value)
-					{
+            public ValueHolder(long maxValue)
+            {
+                MaxValue = maxValue;
+            }
+        }
+
+        public long NextId()
+        {
+            var current = valueHolder;
+            var incrementedCurrent = Interlocked.Increment(ref current.Current);
+
+            while (incrementedCurrent > current.MaxValue)
+            {
+                lock (generatorLock)
```

Comment:
We probably don't need this, given that we now hold all of this in the ValueHolder